app/controllers/application_controller.rb

```diff
@@ -291,6 +291,15 @@ def logstash_catchall
     end
   end
 
+  def enable_replica
+    begin
+      ActiveRecord::Base.connection.enable_replica
+      yield
+    ensure
+      ActiveRecord::Base.connection.disable_replica
+    end
+  end
```

Comment:
Just so I understand, is this turning on read/write splitting? I.e. if an action has this around it, Makara will perform SELECTs in the replica and INSERTs in the primary?

Reply:
Yes. This is a toggle I added for temporary use where we can opt-in a request to read/write splitting. If a controller/action does not call this with `prepend_around_action` then all its queries will run against the primary. If it does call this, it'll use read/write splitting that I plan to configure with "stickiness". Meaning all reads to to the replica until there's an INSERT or UPDATE or anything else that is known to need the primary. After a query goes to the primary (unless we temporarily ignore stickiness), all subsequent queries including SELECTS will go to the primary for the duration of the request, as well as for subsequent requests from the same user up until `primary_ttl` seconds (I plan to make that 5)